src/net/socket.rs

```diff
@@ -31,55 +35,78 @@ impl SocketSystem {
    }
 
    /// Resolves the socket addresses the given hostname could refer to.
-   async fn resolve_address_with_default_port(hostname: &str) -> anyhow::Result<Vec<SocketAddr>> {
-      if let Ok(addresses) = lookup_host(hostname).await {
-         Ok(addresses.collect())
+   async fn resolve_address_with_default_port(hostname: &str) -> anyhow::Result<url::Url> {
```

Comment:
I think `url::Url` could be `use`d to prevent the stutter. This should be simple as using Rust Analyzer's "Replace qualified path with use" assist. Also, the name of the parameter `hostname` isn't accurate anymore and should be renamed to `url`.